apps/rehabstod/src/store/api.ts

```diff
@@ -44,13 +44,14 @@ export const api = createApi({
         }
       },
     }),
-    updateUserPreferences: builder.mutation<UserPreferences, { standardenhet: string }>({
-      query: ({ standardenhet }) => ({
+    updateUserPreferences: builder.mutation<UserPreferences, UserPreferences>({
+      query: (preferences) => ({
         url: 'user/preferences',
         method: 'POST',
-        body: { standardenhet },
+        body: preferences,
       }),
       transformResponse: (response: { content: UserPreferences }) => response.content,
+      invalidatesTags: ['User', 'SickLeaves', 'SickLeaveSummary', 'SickLeavesFilter'],
     }),
```

Comment:
CareProvider uses this and needs to be updated with something like

```typescript
if (user && isChecked && selectedUnit) {
  UpdateUserPreferences({
    ...user.preferences,
    standardenhet: selectedUnit.id,
  })
}
```

And `standardenhet` needs to be updated in the schema: `standardenhet: z.nullable(z.string())`